### PR TITLE
[ONNX] Frontend refactoring

### PR DESCRIPTION
**Pull request by gkrivor**
### Details:
 - Removed an old exported symbols
 - Removed onnx.hpp
 - Removed onnx_importer_visibility.hpp
 - Removed onnx_utils.hpp

### Tickets:
 - 125501

### CODE REVIEW

File: src/frontends/onnx/frontend/include/onnx_import/core/node.hpp
Comment (ilya-lavrenov) [inline src/frontends/onnx/frontend/include/onnx_import/core/node.hpp:1]: if this file is not a public API anymore, we need to move it to `src` folder.

Similar for other files in `include/onnx_import` folder.

Comment (gkrivor) [reply src/frontends/onnx/frontend/include/onnx_import/core/node.hpp:1]: Agree, I’ll do it in a separate pr

Comment (ilya-lavrenov) [reply src/frontends/onnx/frontend/include/onnx_import/core/node.hpp:1]: why not in this one?

**Review by ilya-lavrenov — APPROVED** [review]
Agreed to move old API headers to `src` folder in a separate PR

**Pull request merged**